Update docs on QA and review process (#4926)
geoserver/geoserver: doc/en/developer/source/policies/review.rst

> * *Presence of a proposal*, if required. For any large change our process demands a formal proposal to be made and discussed in the community before a pull request gets made. If the pull request is made anyways, point them to the process and warn that the community might eventually request changes in the approach

Reviewer: ```suggestion
* *Presence of a proposal*, if required. For any large change our process requires a formal proposal to be made and discussed in the community before a pull request gets made. If the pull request is made anyway, point the submitter to the process and note that the community might eventually request changes in the approach.
```

  
  *Where possible we use checkstyle to double check copyright headers are included, you will still need to check xml files during review.*

* *No commented out code sections*. The version control is able to tell differences between existing and past versions of a certain file, thus, the commit should not contain commented out code sections.

* *Javadocs and comments*. Public classes and methods should have a minimum of javadoc (a simple sentence explaning what the method or class does will be sufficient), difficult parts of the code should have some comments explaining what the code does (how it does it, is evident by the code). Comments should be professional (no personal opinions or jokes), current to the code (no need to explain what was there before, unless there is a high risk of it coming back), with no reference to the comment author (in case we need to know that information, a git blame will do).

  *Where possible our build process uses tools like checkstyle to double check javadocs are formatted correctly.*

> * *Code formatting*. The project is setup to reformat code using google code formatter, for every other IDEs it's basically the official Java coding conventions (spaces instead of tabs) with "long" lines.

Reviewer: Maybe: "for other IDEs"

Reviewer 2: Also, the Google Code convention is stricly 80 lines. Not long ones.

Reviewer 2: Maybe  Jody meant that's it's often the default coding convention? Except it's not, one has to install a specific plugin to make it adhere to our standard. And it's not even necessary, building with Maven formats the files to the expected output.

Author: This was mostly the previous text, what should we write here?

> * *Code formatting*. The project is setup to reformat code using google code formatter, for every other IDEs it's basically the official Java coding conventions (spaces instead of tabs) with "long" lines.

Reviewer 2: ```suggestion
* *Code formatting*. The project is setup to reformat code using Google code formatter, for every other IDEs it's basically the official Java coding conventions (spaces instead of tabs) with "long" lines.
```

Author: So what should be a quick summary of “google code formatter”?

> [Google Java style guide](https://google.github.io/styleguide/javaguide.html  ) is basically the official java coding conventions with some clarifications (UTF-8, spaces instead of tabs, no wildcard import, ...) for a consistent document.

Author: Or should we just avoid providing a summary and ask folks to set up google code formatter for their IDE? The [README.md](https://github.com/google/google-java-format) has guidance.

Reviewer: I think its actually the AOSP code formatter. Google is a bit different in terms of (at least) 2 vs 4 spaces.

> * *Code formatting*. The project is setup to reformat code using google code formatter, for every other IDEs it's basically the official Java coding conventions (spaces instead of tabs) with "long" lines.

Reviewer: ```suggestion
* *Code formatting*. The project is set up to reformat code using the AOSP code formatting conventions. This will happen automatically when building with Maven. Some IDEs can do the same formatting, however do a command line Maven build if in doubt. Note that QA checks will fail if there are formatting issues.
```

>   *Where possible our build process reformats the code automatically to reduce frustration during the review process.*

Reviewer: ```suggestion
```
